### DISCUSSION TITLE
Port to Atmega328?

### DISCUSSION THREAD

**Discussion by matthew-quirke** [Ralim/IronOS#1965] [Ideas]
My primary Iron died a few days ago and I had to replace the microcontroller with a Arduino Nano, a Mosfet and some custom code to get back up and going again.

This led to wonder if it might be possible to port IronOS code to an Atmega328? (Arduino Nano)

As I am a total amateur hack, I was wondering if there are/might be any obvious insurmountable problems that I have not foreseen until investing quite a bit of time only to find out it's not possible..

I had a quick go copying the libraries over to a PlatformIO project and compiling - but got hit with a string of errors which I started working through .

Many thanks for ideas - input

**Comment by Ralim** [comment]
Hia,
Realistically the answer for any ATMega is going to be a no.
While its maybe _technically_ doable its not going to be an easy path, and I wont be able to support you with this either (as I haven't used an Atmel/Microchip part in forever).

The main blocker will be the small flash coupled with the tiny RAM; so you wont be able to use font compression (due to tiny ram) to fit it in flash and thus be stuck with larger font data that likely wont fit in flash.
